src/dune_engine/action_exec.ml

```diff
@@ -449,6 +459,13 @@ and exec_list ts ~display ~ectx ~eenv =
     | Need_more_deps _ as need -> Fiber.return need
     | Done -> exec_list rest ~display ~ectx ~eenv)
 
+and exec_list_concurrent ts ~display ~ectx ~eenv =
+  Fiber.parallel_map ts ~f:(fun t ->
+      exec t ~display ~ectx ~eenv >>= function
+      | Need_more_deps _ as need -> Fiber.return need
```

Comment:
This pattern match is not useful, it seems

Reply:
Yep, I've inlined the entire thing anyway.